File: src/java/jvm/api/concurrent/JavaInterruptedError.java
package jvm.api.concurrent;

import jvm.api.result.JavaResults;
import magma.api.concurrent.InterruptedError;

public record JavaInterruptedError(InterruptedException exception) implements InterruptedError {
    @Override
    public String display() {
        return JavaResults.convertThrowableToString(exception);
    }
}

File: src/clang/magma/JavaInterruptedError.c
package magma;class package magma;{package magma;

import magma.error.Error;class import magma.error.Error;{

import magma.error.Error;
import jvm.result.Results;class import jvm.result.Results;{
import jvm.result.Results;

public record JavaInterruptedError(InterruptedException error) implements Error {
    @Override
    public String display() {
        return Results.printStackTraceToString(error);
    }
}
class public record JavaInterruptedError(InterruptedException error) implements Error {
    @Override
    public String display() {
        return Results.printStackTraceToString(error);
    }
}{

public record JavaInterruptedError(InterruptedException error) implements Error {
    @Override
    public String display() {
        return Results.printStackTraceToString(error);
    }
}
